pygmt/datasets/earth_deflection.py

```diff
@@ -0,0 +1,117 @@
+"""
+Function to download the IGPP Earth east-west and south-north deflection datasets from
+the GMT data server, and load as :class:`xarray.DataArray`.
+
+The grids are available in various resolutions.
+"""
+
+from collections.abc import Sequence
+from typing import Literal
+
+import xarray as xr
+from pygmt.datasets.load_remote_dataset import _load_remote_dataset
+
+__doctest_skip__ = ["load_earth_deflection"]
+
+
+def load_earth_deflection(
+    resolution: Literal[
+        "01d", "30m", "20m", "15m", "10m", "06m", "05m", "04m", "03m", "02m", "01m"
+    ] = "01d",
+    region: Sequence[float] | str | None = None,
+    registration: Literal["gridline", "pixel", None] = None,
+    direction: Literal["east_west", "south_north"] = "east_west",
```

Comment:
I think it should be "north_south" rather than "south_north", mainly because:

1. East and North are pointing to the positive values
2. The GMT dataset names are "edefl" and "ndefl".
3. The original source files are named as "east_xxx" and "north_xxx"
```suggestion
    direction: Literal["east_west", "north_south"] = "east_west",
```

I think we should also fix "south-north" to "north-south" in the remote-dataset page.

Reply:
Yes, I am also a bit wondering about the dataset names and coordinate axes directions! And currently looking at in the literature about this naming.

Reply:
Just saw that the remote dataset docs are actually not consistent, and both "south-north" 
(title and image describtion https://www.generic-mapping-tools.org/remote-datasets/earth-ndefl.html#igpp-earth-south-north-deflections) and "north-south" (in section usage https://www.generic-mapping-tools.org/remote-datasets/earth-ndefl.html#usage) are used. 

Wikipedia says is's "east-west" and "north-south" (https://en.wikipedia.org/wiki/Vertical_deflection) and here https://geodesy.noaa.gov/GEOID/GSVS/deflection-vertical.shtml the deflection of the vertical is described via "east" and "north" components. So I think, as suggested by @seisman "north-south" is correct and we should update the remote dataset docs. For the PyGMT side, I am wondering, maybe `component` is better than `direction` for the parameter name.

Reply:
Maybe we should at a note in the docs, that positive values mean a deflection to the east (north) and negative to the west (south) for the `edefl` (`ndefl`) dataset; similar as we did it for the "distance to shoreline" dataset.

| east-west (edefl) | north-south (ndefl) |
| --- | --- |
| ![edefl_grayc](https://github.com/user-attachments/assets/f2b7d436-46d6-4c5e-a18e-4798bf3e62d9) | ![ndefl_grayc](https://github.com/user-attachments/assets/efd1feb4-a8f2-4683-9a61-6554395e5673) |

Reply:
> For the PyGMT side, I am wondering, maybe `component` is better than `direction` for the parameter name.

Yes, I think so.



> Maybe we should at a note in the docs, that positive values mean a deflection to the east (north) and negative to the west (south) for the `edefl` (`ndefl`) dataset; similar as we did it for the "distance to shoreline" dataset.

Sounds good.